### PR TITLE
Use finetune_right_pad_id token if available in KFTO SFT example

### PR DESCRIPTION
**Pull request by astefanutti**
## Description

This PR updates the LLM fine-tuning with KFTO example to use `<|finetune_right_pad_id|>` as padding token for compatible model tokenizers, like Llama 3.1 Instruct.

This is needed to distinguish padding token from EOS token, otherwise inference gets unbounded.

## How Has This Been Tested?

Successfully run the KFTO SFT example.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by ChughShilpa — APPROVED** [review]
/lgtm

**Review by Fiona-Waters — APPROVED** [review]
